robosat/osm/road.py

```diff
@@ -0,0 +1,143 @@
+import sys
+
+import math
+import osmium
+import geojson
+import shapely.geometry
+
+from robosat.osm.core import is_polygon
+
+
+class RoadHandler(osmium.SimpleHandler):
+    """Extracts road polygon features (visible in satellite imagery) from the map.
+    """
+
+    #  highway=* to discard because these features are not vislible in satellite imagery
+    @staticmethod
+    def road_filter():
+        return {
+            "motorway",
+            "trunk",
+            "primary",
+            "secondary",
+            "tertiary",
+            "unclassified",
+            "residential",
+            "service",
+            "motorway_link",
+            "trunk_link",
+            "primary_link",
+            "secondary_link",
+            "tertiary_link"
+        }
```

Comment:
I think this could be an attribute (set) instead of a static method.

Reply:
In the programming, the IDE hints that "This inspection detects call for function "set" which can be replaced with set literal". So I choose the set literal method to construct set.

Reply:
There is something wrong with the geojson.io. Every time I open the geojson file(about 40.7 MB), it crashed. May I send the file directly to you(I use QGIS to check the file)?

Reply:
I only upload data at "https://gist.github.com/DragonEmperorG/666ee2304df667e63519e0a93a61707e"

Reply:
Here's your GeoJSON overlayed on top of the Mapbox satellite layer using Mapbox Studio:

https://api.mapbox.com/styles/v1/danieljh/cjlxi6qmy4qad2rnzio16pklo.html?fresh=true&access_token=pk.eyJ1IjoiZGFuaWVsamgiLCJhIjoiTnNYb25JSSJ9.vYOnsuu1zeKcGW2nj0uJZw#12/30.54169/114.25045

Not sure why your data only contains a couple of linestring features and not all of them for roads. It would be great to spot-check different types of roads - from skimming the map for highways and larger roads it definitely looks okay.

Reply:
This time, I use Mapbox Studio to create a map, and this is the result.
https://api.mapbox.com/styles/v1/dragonemperormapbox/cjlyfpts55cs92sqr7s14tbbh.html?fresh=true&title=true&access_token=pk.eyJ1IjoiZHJhZ29uZW1wZXJvcm1hcGJveCIsImEiOiJjamplMDU3MHM0anU0M3FvNjZmZmE2ODBiIn0.5kp8np7mfkXk0TWmGUHePg#13.0/33.750013/-118.410600/0

Reply:
Thanks; this looks good overall. There are some problems with the OpenStreetMap data like [here](https://api.mapbox.com/styles/v1/dragonemperormapbox/cjlyfpts55cs92sqr7s14tbbh.html?fresh=true&title=true&access_token=pk.eyJ1IjoiZHJhZ29uZW1wZXJvcm1hcGJveCIsImEiOiJjamplMDU3MHM0anU0M3FvNjZmZmE2ODBiIn0.5kp8np7mfkXk0TWmGUHePg#16.39/30.526138/114.180132) but that's not related to this pull request. I'm happy with how it turned out in general!